### PR TITLE
Add option for alternate graph planning algorithm; modifications to scoring and visualization

### PR DESCRIPTION
**Pull request by shuail**
Based on our needs, I modify the code to output more information (score, filename, connectivity etc) and add more similarity scores (tanimoto score), I also try to change some threshold based on the experiences of running the code towards our test set. For graphing part, I add couple radial options to create radial shape graphs. Below are the details: 

1, add erc score (charge score) to DBMolecules database (dbmol.py)
2, change the total charge comparison threshold from 1e-3 to 1e-2 (dbmol.py)
3, add a function to calculate the tanimoto score (mcs.py)
4, add radial graphing option (in both dbmol.py and graphgen.py):
    Complete radial option will pick the radial center (lead compound) automatically by the structure similarity: the ligand shared the most structural similarity with others will be picked as lead compounds
    Biased radial option will take the custom specified compound as the lead compound.
5, output the pickle file containing DBMolecules database with graph info. (dbmol.py)
6, change the max_mol_size from 11 to 50 angstrom in the graphing to accommodate larger molecule. (graphgen.py)
7, improve the drawing function to layout the 2D structure. (graphgen.py) 
8, add the layout_info function in the graphgen.py to output ligand index, filename, multiple scores and connectivities (graphgen.py)

### CODE REVIEW

**Comment by davidlmobley** [comment]
@shuail - we discussed doc strings a bit. Did you have any updates to those before I begin reviewing?

Also, can you explain what "erc score" is, in your description above?

**Comment by shuail** [comment]
@davidlmobley Oops, that's my fault, it should be ecr score which is corresponding to the ECR (EleCtrostatic Rule) in the original code. I will push a change to fix that and add some documentations in the README file to explain how to use the radial options. All should be done by today.

**Comment by shuail** [comment]
@davidlmobley fix some bugs and add more documentations in the README file to explain the radial graphing.

**Comment by davidlmobley** [comment]
Looks like Travis tests are failing for something which is not your fault, @shuail  -- I think @nividic had to build a graphviz for travis to use and it looks like it may have gone out of date. Is this something you want to look into or should I see if I can sort it out?

File: README.md [hunk 1]
```diff
@@ -62,6 +62,13 @@ import lomap
 
 db_mol = lomap.DBMolecules("python string pointing to a directory with mol2 files", output=True)
 
+    #More graphing options:
+    # Use the complete radial graph option. The radial center (lead compound) is automatically picked by the structure similarity: the ligand shared the most structural similarity with others will be picked as lead compounds
```
Comment (davidlmobley) [inline README.md]: Minor grammar stuff here; I can clean it up after merge if needed. It would be something like "... the ligand with the most structural similarity to all of the others will be picked as 
 the 'lead compounds' and used as the central compound..." or something along those lines.

File: README.md [hunk 2]
```diff
@@ -62,6 +62,13 @@ import lomap
...
+    db_mol = lomap.DBMolecules("python string pointing to a directory with mol2 files", output=True, radial=True)
+
+    # Use the biased radial graph option. The radial center (lead compound) will be the filename specified by user.  
+    db_mol = lomap.DBMolecules("python string pointing to a directory with mol2 files", output=True, radial=True, bias="python string pointing to the mol2 filename")
```
Comment (davidlmobley) [inline README.md]: I'm not totally sure I'm happy with this terminology -- bias sounds like a type of weighting to me, which this is not. Do you want to maybe call this something like `hub_compound` rather than `bias`? So this would be called "Use a radial graph with a manually specified hub compound", and the argument would be `hub=filename.mol2` or similar.

File: README.md (hunk 1 above)
Comment (davidlmobley) [reply README.md]: In keeping with my terminology suggestion below, if it were me I would rephrase this as "the ligand with the most structural similarity to all of the others will be picked as the 'hub compound'" and I would call this "Use a radial graph approach where the central (hub) compound is automatically selected."

File: lomap/dbmol.py [hunk 3]
```diff
@@ -125,21 +126,27 @@ def __init__(self, directory, parallel=1, verbose='off',
             if not isinstance(display,  bool):
                 raise TypeError('The display flag is not a bool type')
 
+            if not isinstance(radial,  bool):
+                raise TypeError('The display flag is not a bool type')
```
Comment (davidlmobley) [inline lomap/dbmol.py]: I think your Exception is wrong here. Shouldn't it say, "The radial flag is not a bool type"?

File: lomap/dbmol.py [hunk 4]
```diff
@@ -68,10 +68,11 @@ class DBMolecules(object):
     """
 
     # Initialization function
+    #sliu 07/17 add the radial graph option
```
Comment (davidlmobley) [inline lomap/dbmol.py]: Don't store history about what you changed and when into the code itself. GitHub keeps track of this for you. No need for this type of comment.

File: lomap/dbmol.py [hunk 5]
```diff
@@ -125,21 +126,27 @@ def __init__(self, directory, parallel=1, verbose='off',
             if not isinstance(display,  bool):
                 raise TypeError('The display flag is not a bool type')
 
+            if not isinstance(radial,  bool):
+                raise TypeError('The display flag is not a bool type')
             output_str=''
             display_str=''
+            radial_str=''
```
Comment (davidlmobley) [inline lomap/dbmol.py]: Set these to `None` initially? e.g. `radial_str = None`? Just asking. (I realize the first two of these are not your code.)

File: lomap/dbmol.py [hunk 6]
```diff
@@ -379,8 +391,8 @@ def ecr(mol_i, mol_j):
             
             for atom in mol_j.GetAtoms():
                 total_charge_mol_j += float(atom.GetProp('_TriposPartialCharge'))
-
-            if abs(total_charge_mol_j - total_charge_mol_i) < 1e-3:
+            #sliu 07/17 change to a loose threshold
```
Comment (davidlmobley) [inline lomap/dbmol.py]: No need for this comment.

File: lomap/dbmol.py [hunk 7]
```diff
@@ -379,8 +391,8 @@ def ecr(mol_i, mol_j):
...
+            if abs(total_charge_mol_j - total_charge_mol_i) < 1e-2:
```
Comment (davidlmobley) [inline lomap/dbmol.py]: Should the threshold here be an optional argument to this function? I'm fine to having it default to your new behavior, just asking -- will there ever be cases where we need to change this?

File: lomap/dbmol.py [hunk 8]
```diff
@@ -435,12 +447,12 @@ def ecr(mol_i, mol_j):
             # The total score will be the product of all the single rules
                
             tmp_scr = ecr_score * MC.mncar() * MC.mcsr()
-            
             strict_scr = tmp_scr *  MC.tmcsr(strict_flag=True) 
             loose_scr = tmp_scr * MC.tmcsr(strict_flag=False) 
-        
+            #tanimoto = MC.mtansr()
```
Comment (davidlmobley) [inline lomap/dbmol.py]: Why do we have something commented out here? If you don't need it just remove it.

File: lomap/dbmol.py [hunk 9]
```diff
@@ -458,13 +470,14 @@ def build_matrices(self):
         # which implements a basic class for symmetric matrices
         self.strict_mtx = SMatrix(shape=(self.nums(),))
         self.loose_mtx = SMatrix(shape=(self.nums(),))
-
+        #sliu 07/17 add this to output information about ligand charges
```
Comment (davidlmobley) [inline lomap/dbmol.py]: Remove dated change-log comments.

File: lomap/dbmol.py [hunk 10]
```diff
@@ -534,6 +545,9 @@ def build_graph(self):
         if self.options.output:
             try:
                 Gr.writeGraph()
+                #sliu 07/17 change here to output the object as pickle file
```
Comment (davidlmobley) [inline lomap/dbmol.py]: Changelog not needed.

File: lomap/dbmol.py [hunk 11]
```diff
@@ -901,10 +915,9 @@ def startup():
     ops= parser.parse_args()
     
     # Molecule DataBase initialized with the passed user options
-    db_mol = DBMolecules(ops.directory, ops.parallel, ops.verbose, ops.time, ops.ecrscore, 
-                        ops.output, ops.name, ops.display, ops.max, ops.cutoff) 
-
-   
+    #sliu 07/17 change to add the radial option
```
Comment (davidlmobley) [inline lomap/dbmol.py]: -changelog

File: lomap/dbmol.py [hunk 12]
```diff
@@ -954,6 +967,11 @@ def startup():
                          help='The maximum distance used to cluster the graph nodes')
 graph_group.add_argument('-c', '--cutoff', default=0.4 , action=check_cutoff, type=float,\
                          help='The Minimum Similariry Score (MSS) used to build the graph')
+#sliu 07/17 add radial option
```
Comment (davidlmobley) [inline lomap/dbmol.py]: -changelog

File: lomap/graphgen.py [hunk 13]
```diff
@@ -92,6 +92,11 @@ def __init__(self, dbase):
 
         self.similarityScoresLimit = dbase.options.cutoff
        
+        #sliu 07/17 add radial option of graphing
```
Comment (davidlmobley) [inline lomap/graphgen.py]: -changelog

File: lomap/graphgen.py [hunk 14]
```diff
@@ -106,7 +111,8 @@ def __init__(self, dbase):
         self.max_nodes = 100
 
         # The maximum threshold distance in A unit used to select if a molecule is depicted
```
Comment (davidlmobley) [inline lomap/graphgen.py]: Not your fault, but can we make this "angstroms" rather than "A"? It took me a minute to figure out what this was saying. (Actually, I still don't totally understand it -- is it saying that molecules that are more than `max_mol_size` angstroms in some dimension won't be depicted?)

File: lomap/graphgen.py [hunk 15]
```diff
@@ -106,7 +111,8 @@ def __init__(self, dbase):
...
-        self.max_mol_size = 11.0
+        #sliu 07/17 switch to a larger size
```
Comment (davidlmobley) [inline lomap/graphgen.py]: -changelog

File: lomap/graphgen.py [hunk 16]
```diff
@@ -166,6 +172,31 @@ def __init__(self, dbase):
 
         return
 
+    #sliu 07/17 add function to pick the lead compounds for the radial graphing option
```
Comment (davidlmobley) [inline lomap/graphgen.py]: -changelog

File: lomap/graphgen.py [hunk 17]
```diff
@@ -166,6 +172,31 @@ def __init__(self, dbase):
...
+    def pick_lead(self):
+        if (self.dbase.nums() * (self.dbase.nums() - 1)/2) != self.dbase.strict_mtx.size:
```
Comment (davidlmobley) [inline lomap/graphgen.py]: Can you add a comment saying how you know there is a problem if this condition is met? It's not immediately obvious to me.

File: lomap/graphgen.py [hunk 18]
```diff
@@ -166,6 +172,31 @@ def __init__(self, dbase):
...
+            raise ValueError("There are errors in the similarity score matrices")
+        if not self.dbase.options.bias == "None":
+            #bias radial option
```
Comment (davidlmobley) [inline lomap/graphgen.py]: This would be a good place to recap what this does as opposed to just naming it (and likewise on line 189 in the same file for "complete radial option". Maybe here you could say "Use the provided reference compound as a hub" and below you could say, "Pick the compound with the highest total similarity to all other compounds to use as a hub" or something...?

File: lomap/graphgen.py [hunk 19]
```diff
@@ -332,12 +362,16 @@ def minimizeEdges(self):
             if len(subgraph.edges()) > 2:   # Graphs must have at least 3 edges to be minimzed
 
                 for edge in weightsList:
-
-                    subgraph.remove_edge(edge[0], edge[1])
-
-                    if self.checkConstraints(subgraph, numberOfComponents) == False:
-                        subgraph.add_edge(edge[0], edge[1], similarity = edge[2], strict_flag = True)
-
+                    #sliu 07/17 add radial option here to avoid removing edges connect to the lead compound
```
Comment (davidlmobley) [inline lomap/graphgen.py]: Changelog not necessary, though this may be a good place to have a comment saying what you are doing! The current comment is incomprehensible to me.

Comment (davidlmobley) [reply lomap/graphgen.py]: Also, it looks like these changes here are applied always, rather than only if it is a radial option? So you're altering the prior behavior of this code, right? Is that intentional or did you forget a conditional here?

File: lomap/graphgen.py [hunk 20]
```diff
@@ -726,21 +760,22 @@ def max_dist_mol(mol):
...
-                    
-                    mol = AllChem.RemoveHs(mol)
-        
-                    AllChem.Compute2DCoords(mol)
-                
                     fname = os.path.join(directory_name, self.dbase[id_mol].getName() + ".png")
-                    
-                    Draw.MolToFile(mol, fname, size=(100,100), fitimage=True, imageType='png' )
-                    
+                    #sliu 07/17
```
Comment (davidlmobley) [inline lomap/graphgen.py]: changelog not necessary -- except that you want to make sure you explain exactly what you did here and why in the notes on the PR. Why are you changing this?

File: lomap/graphgen.py [hunk 21]
```diff
@@ -726,21 +760,22 @@ def max_dist_mol(mol):
...
-                    
+                    #sliu 07/17 
+                    #1, modify here to calculate the 2D structure for ligands cannot remove Hs by rdkit
+                    #2, change the graph size to get better resolution            
+                    try:
+                        mol = AllChem.RemoveHs(mol)
+                        AllChem.Compute2DCoords(mol)
+                        Draw.MolToFile(mol, fname, size=(200,200), kekulize=False, fitimage=True, imageType='png' )
+                    except:
+                        logging.info("This molecule %s has trouble to remove Hs using rdkit"%self.dbase[id_mol].getName())
```
Comment (davidlmobley) [inline lomap/graphgen.py]: What's "Hs"? Hydrogens? Maybe better, "Error attempting to remove hydrogens for molecule %s using RDKit." (?) or similar.

What kind of "trouble"?

File: lomap/graphgen.py [hunk 22]
```diff
@@ -755,8 +790,32 @@ def max_dist_mol(mol):
         os.system(cmd)
         os.remove(self.dbase.options.name+'_tmp.dot')
         shutil.rmtree(directory_name, ignore_errors=True)
-
-        
+    #sliu 07/27 add function to output the score and connectivity txt file
```
Comment (davidlmobley) [inline lomap/graphgen.py]: dated changelog not necessary -- though comments explaining what is being done here are always good.

File: lomap/graphgen.py [hunk 23]
```diff
@@ -770,12 +829,15 @@ def writeGraph(self):
 
         try:
             self.dbase.write_dic()
+            #sliu 07/17 add layout_info function
```
Comment (davidlmobley) [inline lomap/graphgen.py]: -changelog

File: lomap/graphgen.py [hunk 24]
```diff
@@ -770,12 +829,15 @@ def writeGraph(self):
...
+            self.layout_info()
         except Exception as e:
             raise IOError("%s: %s.txt" % (str(e), self.dbase.options.name))
  
         try:
             self.generate_depictions()
             nx.nx_agraph.write_dot(self.resultGraph, self.dbase.options.name+'.dot')
+            #print "Check the number of nodes %s"%(nx.number_of_nodes(self.resultGraph))
```
Comment (davidlmobley) [inline lomap/graphgen.py]: Shouldn't have unused code here.

File: lomap/graphgen.py [hunk 25]
```diff
@@ -901,15 +963,23 @@ def max_dist_mol(mol):
             for each_node in self.resultGraph:
                 
                 id_mol = self.resultGraph.node[each_node]['ID']
-                mol = AllChem.RemoveHs(self.dbase[id_mol].getMolecule())
+                #sliu 07/17 skip remove Hs by rdkit if Hs cannot be removed
```
Comment (davidlmobley) [inline lomap/graphgen.py]: No changelog. Though, this seems important. What will be the result if this is skipped? Why does it happen? Are we trying to get the RDKit people to fix the problem, if it is on their end?

File: lomap/graphgen.py [hunk 26]
```diff
@@ -901,15 +963,23 @@ def max_dist_mol(mol):
...
+                    mol = self.dbase[id_mol].getMolecule()
             
                 # max_dist = max_dist_mol(mol)
                 # if max_dist > 7.0:
                 #     continue
 
                 AllChem.Compute2DCoords(mol)
-                
-                img_mol = Draw.MolToImage(mol,mol_size)
+                #sliu 07/17 add try exception for cases cannot be draw
```
Comment (davidlmobley) [inline lomap/graphgen.py]: -changelog
But might be useful to retain a comment explaining what's going on and why.

File: lomap/mcs.py [hunk 27]
```diff
@@ -422,6 +424,12 @@ def getMapping(moli, molj, hydrogens=False, fname=None, time_out=150):
     ############ MCS BASED RULES ############
 
     # MCSR Rule
+    #sliu 0/17 add function to calculate tanimoto
+    def mtansr(self,):
```
Comment (davidlmobley) [inline lomap/mcs.py]: This is unused in the code. Remove? Or note that it is not used but is being retained in case you want to use it in the future?

**Review by davidlmobley — CHANGES_REQUESTED** [review]
@shuail - in general looks good, but I think a few small tweaks are needed:
- As noted elsewhere, we really need some unit tests in here (though this could come in in a separate PR if needed; let me know what you want to do in this respect)
- Ultimately, it would be easier to explain/document the new options available here with the ability to display interesting example graphs, so see #25 
- I provided some comments on cleaning up your code a bit, documenting better, etc. In general you don't need to maintain a list of revisions in your code itself, as GitHub does that for you (and adding them to your code just makes it harder to read).
- Currently the travis tests are failing because of a dependency, it seems, so we need to sort out who will deal with that

Let me know which of these you want to address in the same PR versus which we should address separately after merging this one. Really all that HAS to be done here before merging, I think, is the code cleanup stuff.

**Comment by shuail** [comment]
@davidlmobley Thanks for the checking and comments, I appreciate for that! I will try to cleanup the code based on comments here today or tomorrow and add a example case for the new graphing option. The next step would be add more unit tests in the test code. I am ok with either 1, merging right after the clean up or 2, after the new unit test being added. For the travis issue, I am not sure how to fix that instantly. So it'll be great if you have some ideas to easily bypass that otherwise, I will start to look into some details about travis. Thx!

**Comment by davidlmobley** [comment]
Regarding Travis -- it's not an issue with travis specifically. Basically, Travis is just trying to install all the prerequisites itself (usually via conda) and then run our specified tests. It's having trouble installing one of the dependencies, so it fails. Presumably anyone else installing in the same, straightforward way that Travis is doing (see https://github.com/MobleyLab/Lomap/blob/master/.travis.yml and https://github.com/MobleyLab/Lomap/blob/master/devtools/conda-recipe/meta.yaml) will run into the same error.

**Comment by shuail** [comment]
@davidlmobley yes, I get some errors after running "conda build devtools/conda-recipe" under [https://github.com/MobleyLab/Lomap/blob/master/.travis.yml](url) The error is below: 

ERROR:conda.core.link:An error occurred while installing package 'nividic::graphviz-2.38.0-1'.
PaddingError: Placeholder of length '80' too short in package /home/shuai/miniconda/conda-bld/lomap_1501794001217/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/graphviz/libgvplugin_pango.so.6.0.0.

File: README.md (hunk 1 above)
Comment (shuail) [reply README.md]: I agree, will change the readme file and option name accordingly

File: lomap/dbmol.py (hunk 4 above)
Comment (shuail) [reply lomap/dbmol.py]: Got it, will remove all similar logs.

File: lomap/dbmol.py (hunk 3 above)
Comment (shuail) [reply lomap/dbmol.py]: Yes, you are right. It's my fault, will change to the "The radial flag is not a bool type"

File: lomap/dbmol.py (hunk 5 above)
Comment (shuail) [reply lomap/dbmol.py]: Based on the code below I think later the radial_str will be appended to the names_str and an empty string would be good to indicate that the option is not turn on. So I think empty string is ok here.

File: lomap/dbmol.py (hunk 7 above)
Comment (shuail) [reply lomap/dbmol.py]: I think it'll not be changed for most of the time, so it's fine to set a fixed default value here I guess.

File: lomap/mcs.py (hunk 27 above)
Comment (shuail) [reply lomap/mcs.py]: yes, just comments to retained for future use.

File: lomap/dbmol.py (hunk 8 above)
Comment (shuail) [reply lomap/dbmol.py]: Yes, removed

File: lomap/dbmol.py (hunk 9 above)
Comment (shuail) [reply lomap/dbmol.py]: removed

File: lomap/dbmol.py (hunk 10 above)
Comment (shuail) [reply lomap/dbmol.py]: removed

File: lomap/dbmol.py (hunk 11 above)
Comment (shuail) [reply lomap/dbmol.py]: remvoed

File: lomap/dbmol.py (hunk 12 above)
Comment (shuail) [reply lomap/dbmol.py]: remvoed

File: lomap/graphgen.py (hunk 13 above)
Comment (shuail) [reply lomap/graphgen.py]: remvoed

File: lomap/graphgen.py (hunk 14 above)
Comment (shuail) [reply lomap/graphgen.py]: yes, for some big molecules with the small max_mol_size, it will not be depicted.

File: lomap/graphgen.py (hunk 15 above)
Comment (shuail) [reply lomap/graphgen.py]: removed

File: lomap/graphgen.py (hunk 16 above)
Comment (shuail) [reply lomap/graphgen.py]: removed

File: lomap/graphgen.py (hunk 17 above)
Comment (shuail) [reply lomap/graphgen.py]: Basically, this line was copied from the function generateInitialSubgraphList() in the previous code. The strict_mtx is an SMatrix object defined in the dbmol.py and the size of this matrix was initialed using shape[0]*(shape[0] - 1)/2. So I think the line here just to ensure that during the calculation the dbase.nums and strict_mtx.size is still consistent with each other.

File: lomap/dbmol.py (hunk 5 above)
Comment (davidlmobley) [reply lomap/dbmol.py]: Ah, OK. Sorry, didn't check that carefully.

File: lomap/graphgen.py (hunk 14 above)
Comment (davidlmobley) [reply lomap/graphgen.py]: So, can you just clarify the comment here, then? :)

File: lomap/mcs.py (hunk 27 above)
Comment (davidlmobley) [reply lomap/mcs.py]: OK, so probably just say "This calculates a Tanimoto similarity of ... for .... Currently this code is not used, but it may be in the future," or something along those lines.

File: lomap/graphgen.py (hunk 19 above)
Comment (shuail) [reply lomap/graphgen.py]: Yes, explain the logic more. For the option always applied question, actually, it'll only turn on when the lead_index is not None. If the radial option is not enabled, the lead_index will be None (default). And if the radial option is on, then the code will try to calculate the lead_index depending on the  complete/biased(hub) option. If it successfully find the lead_index, then the lead_index be passed here.

File: lomap/graphgen.py (hunk 20 above)
Comment (shuail) [reply lomap/graphgen.py]: removed

File: lomap/graphgen.py (hunk 21 above)
Comment (shuail) [reply lomap/graphgen.py]: Change the logging info to "Error attempting to remove hydrogens for molecule %s using RDKit. RDKit cannot kekulize the molecule"

File: lomap/graphgen.py (hunk 22 above)
Comment (shuail) [reply lomap/graphgen.py]: removed the data info

File: lomap/graphgen.py (hunk 23 above)
Comment (shuail) [reply lomap/graphgen.py]: removed

File: lomap/graphgen.py (hunk 24 above)
Comment (shuail) [reply lomap/graphgen.py]: yes, removed

File: lomap/graphgen.py (hunk 19 above)
Comment (davidlmobley) [reply lomap/graphgen.py]: OK. Please do have comments explaining this in the code. :)

File: lomap/graphgen.py (hunk 25 above)
Comment (shuail) [reply lomap/graphgen.py]: When rdkit try to remove hydrogens, it will kekulize the molecule object, while the rdkit will have errors to kekulize molecules (for most cases I saw is the heterocycle). If the RemoveHs is not applied, the kekulized process will be skip so the molecule object will be passed for later layout but in this case, it will have Hydrogens but it looks fine I think.

Comment (shuail) [reply lomap/graphgen.py]: Yes, I think that's the problem at RDKit end, I may give an example in the issue tracker to illustrate this issue.

File: lomap/graphgen.py (hunk 21 above)
Comment (davidlmobley) [reply lomap/graphgen.py]: OK. So we should:
a) document this in the code
b) Give the RDKit people some example molecules and ask them to try to fix the issue
c) Document in the code that we've done so

Then we can revisit this part of the code once the RDKit folks fix it.

File: lomap/graphgen.py (hunk 26 above)
Comment (shuail) [reply lomap/graphgen.py]: Cannot remember if there is any cases catch in the try exception. Maybe this try exception is just for debugging and also let the code bypass the Draw function instead of breaking the for loop. Add more words in the exception log though.

File: lomap/graphgen.py (hunk 14 above)
Comment (shuail) [reply lomap/graphgen.py]: oh, yes, the max_mol_size will compared to the max_disi_mol where defined in the generate_depictions function. So the max_disi_mol return the maximum distance between any two atoms in a molecule. If the molecule is big, the max_disi_mol is large and will exceed the threshold which is defined by max_mol_size and so the molecule will be discard for depicting.

File: lomap/graphgen.py (hunk 18 above)
Comment (shuail) [reply lomap/graphgen.py]: yes, change this lines to add more details

File: lomap/graphgen.py (hunk 21 above)
Comment (shuail) [reply lomap/graphgen.py]: Sounds good to me, will document the code and prepare an example.

**Comment by shuail** [comment]
@davidlmobley  I just commit the code according to my comments and add an example code/folder to show the radial graphs. I will then work on the unit test and prepare an rdkit kekulized failure example.

**Comment by shuail** [comment]
@davidlmobley Some unit tests has been added to cover the radial graph option. Basically, I just follow the original pattern in the test_lomap.py code to ensure that the code will generate identical graphs between 1, the saved graph (generate using the current code) and 2, the graph regenerated by the unit test code. Also I add an problematic case in the issue tracker #26 to show the issue where the rdkit cannot kekulize the molecule. So I think the code is good for you to review. Thanks!

**Review by davidlmobley — APPROVED** [review]
Thanks, @shuail ! Looks good. We could merge; tests pass locally? The remaining issue is that the Travis tests fail; I'll ask Gaetano if he has time to look at the relevant issue with graphviz.

**Comment by shuail** [comment]
@davidlmobley Yes, the code pass the unit test locally and ready to merge I think. Yes, I guess the travis issue is related to the version of graphviz in the Gaetano's channel. If that is not an easy fix, we could consider to rebuild a conda channel? Let me know if you want me to do that, I could chip in and check out more details for that.

**Comment by nividic** [comment]
@shuail @davidlmobley I have update my graphviz conda repository. Travis is not complaining anymore. Unfortunately I tried to install lomap locally and I had some issues. To make it to work I had follow this link https://github.com/awslabs/aws-shell/issues/161. The problem is related to OSX only. Please let me know if you have the same problem or not. On linux it seems to  work.

**Comment by davidlmobley** [comment]
I'm closing and re-opening this PR to re-trigger the tests. @shuail - if the tests pass you should be good to merge.

**Comment by shuail** [comment]
@nividic Thanks for updating the graphviz! @davidlmobley all checks have passed this time!

**Comment by davidlmobley** [comment]
@shuail - tests passed; merging. Thanks!
